### PR TITLE
Examples for ``bad-exception-context``

### PR DESCRIPTION
**Pull request by DudeNr33**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description
Examples and related links for ``bad-exception-context``.

Ref: #5953

### CODE REVIEW

File: doc/data/messages/b/bad-exception-context/bad.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def divide(x, y):
+    result = 0
+    try:
+        result = x / y
+    except ZeroDivisionError:
+        raise Exception("Can't divide by zero!") from result  # [bad-exception-context]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-exception-context/bad.py:6]:
```suggestion
    except ZeroDivisionError:
        raise ValueError(f"Division by zero when dividing {x} by {y} !") from result  # [bad-exception-context]
```

File: doc/data/messages/b/bad-exception-context/good.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+def divide(x, y):
+    result = 0
+    try:
+        result = x / y
+    except ZeroDivisionError as exc:
+        raise Exception("Can't divide by zero!") from exc
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-exception-context/good.py:6]:
```suggestion
    except ZeroDivisionError as exc:
        raise ValueError(f"Division by zero when dividing {x} by {y} !") from exc
```

**Review by Pierre-Sassoulas — APPROVED** [review]
Nitpick to not use an Exception in the re-raise feel free to ignore.

File: doc/data/messages/b/bad-exception-context/bad.py (hunk 1 above)
Comment (DudeNr33) [reply doc/data/messages/b/bad-exception-context/bad.py]: Yes, I was also a bit hesitant using just the general ``Exception`` class. Changed it. Thanks for the feedback!

Comment (Pierre-Sassoulas) [reply doc/data/messages/b/bad-exception-context/bad.py]: Yeah, I think it's bad style to re-reaise an exception that is more generic than the one caught which is why we dont like this, but ValueError do not fix that. Let me suggest something with a custom exception.

File: doc/data/messages/b/bad-exception-context/bad.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+def divide(x, y):
+    result = 0
+    try:
+        result = x / y
+    except ZeroDivisionError:
+        raise ValueError(f"Division by zero when dividing {x} by {y} !") from result  # [bad-exception-context]
+    return result
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-exception-context/bad.py:7]:
```suggestion
class YParameterIsZero(ZeroDivisionError):
    ...

def divide(x, y):
    result = 0
    try:
        result = x / y
    except ZeroDivisionError:
        raise YParameterIsZero(f"Division by zero when dividing {x} by {y} !") from result  # [bad-exception-context]
    return result
```

File: doc/data/messages/b/bad-exception-context/good.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+def divide(x, y):
+    result = 0
+    try:
+        result = x / y
+    except ZeroDivisionError as exc:
+        raise ValueError(f"Division by zero when dividing {x} by {y} !") from exc
+    return result
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/b/bad-exception-context/good.py:7]:
```suggestion
class YParameterIsZero(ZeroDivisionError):
    ...

def divide(x, y):
    result = 0
    try:
        result = x / y
    except ZeroDivisionError as exc:
        raise YParameterIsZero(f"Division by zero when dividing {x} by {y} !") from exc
    return result
```

**Comment by Pierre-Sassoulas** [comment]
Nevermind the suggestion it's not that important :)

**Comment by DanielNoord** [comment]
Oops, sorry guys! Little too trigger happy 😅

**Comment by DudeNr33** [comment]
😀  No problem, while I think Pierre's last suggestion would have been the best solution, I guess it should be already good enough for the sake of documentation.

**Pull request merged**